_posts/2018-08-09-deploying-esxi-through-foreman.md

```diff
@@ -0,0 +1,402 @@
+    <pre><code>
+    def all_loaders_map(precision = 'x64')
+      {
+        "None" => "",
+        "PXELinux BIOS" => "pxelinux.0",
+        "PXELinux UEFI" => "pxelinux.efi",
+        <b>"ESXi PXELinux BIOS" => "pxelinux-3.86.0",</b> # Add this line
+        ...
+    </pre></code>
+
+  * **2.2.2:** Restart httpd/apache2 for the changes to be reflected.
+
+**2.3: Mount the ESXi ISO**  
+We will mount the ESXi ISO under `/var/lib/tftpboot/boot/ESXi-6.7.0-8169922`. Note that this path corresponds to the **KERNEL** entry in the PXELinux template.
+
+  ```sh
+  cd /var/lib/tftpboot/
+
+  mkdir -p boot/ESXi-6.7.0-8169922
+
+  mount ~/Desktop/VMware-VMvisor-Installer-6.7.0-8169922.x86_64.iso boot/ESXi-6.7.0-8169922
+  ```
+**2.4: Edit boot.cfg**  
+Copy the boot.cfg file from the mountpoint to tftproot and edit it to add a prefix to the ISO mountpoint. Make sure to use the same filename as the one specified in the **APPEND** line of the PXELinux template.
+
+  ```sh
+  cd /var/lib/tftpboot/
+
+  cp boot/ESXi-6.7.0-8169922/boot.cfg boot-ESXi-6.7.0-8169922.cfg
+
+  sed -e "s#/##g" -e "3s#^#prefix=/boot/ESXi-6.7.0-8169922\n#" -i boot-ESXi-6.7.0-8169922.cfg
+  ```
+
+  Note: Instead of mounting the ISO, you can also extract the ISO contents to the directory and edit the boot.cfg directly inside that directory. Remember to change the PXELinux template accordingly. I chose this because this approach is a tad cleaner.  
+
+  Now we are ready to deploy this OS on to a host.
+
+**2.5:** Open https://{foreman-url}/hosts/ and edit the host on which you want to deploy ESXi.  
+
+**2.6:** Under the 'Operating System' tab, select the following values:
+
+  Operating System : ESXi 6.7 Build 8169922  
+  Media : ESXi Dummy  
+  Partition table: Kickstart default  
+  PXE Loader : ESXi PXELinux BIOS  
+
+**2.7:** Press 'Resolve' beside Provisioning templates and you should see the ESXi PXELinux and Kickstart templates.
+  <p float='left' align='left'>
+    <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/edit_host_legacy_bios.png" width="400" />
+  </p>
+
+**2.8:** Press 'Build' and reboot the host to PXE. If all goes well, you should see ESXi installer load up and perform an automated installation.
+  <p float='left' align='center'>
+    <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/esxi_installer_loading.png" width="400" />
+  </p>
+
+Now once installation is complete and the host reboots, it will try to PXEBoot and **should** be presented with the default PXE menu.
+<p float='left' align='center'>
+  <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/localboot.png" width="400" />
+</p>
+Upon timeout, the host should attempt a Localboot and boot to ESXi.
+
+However, since the host will boot into the pxelinux.0 v3.86 bootloader, without the supporting files of the same version (menu.c32 etc) the host will not be presented with this PXE menu and will never boot into ESXi.
```

Comment:
Isn't better to put pxelinux v3 into a subdirectory like `pxelinux3/` with all the c32 files and create a relative symlink for `pxelinux3/pxelinux.cfg` pointing to the configuration directory one level above?

Reply:
I had kept it this way to allow for similarity between BIOS and UEFI. But now I realize that it was complicating an otherwise easy task. Have incorporated the changes you requested.